src/main.rs

```diff
@@ -177,188 +225,225 @@ async fn sync(
+        pg_host: env::var("PG_HOST").expect("PG_HOST must be set"),
+        pg_username: env::var("PG_USERNAME").expect("PG_USERNAME must be set"),
+        pg_password: env::var("PG_PASSWORD").expect("PG_PASSWORD must be set"),
+        pg_dbname: env::var("PG_DBNAME").expect("PG_DBNAME must be set"),
+        pg_port: 5432,
+        pg_batch_size: 10000,
+        blaze_page_resource_count: 5000,
+        blaze_num_connection_attempts: 20,
+        target_time: match NaiveTime::from_hms_opt(3, 0, 0) {
+            Some(time) => time,
+            None => {
+                error!("Invalid target time");
+                return Err(anyhow::Error::msg("Invalid target time"));
             }
-        },
-        Err(_) => {
-            eprintln!("PG_BATCH_SIZE must be set. Using default value.");
-            10000 // default value if env var is not set
         }
     };
-    let page_resource_count = match dotenv::var("BLAZE_PAGE_RESOURCE_COUNT") {
-        Ok(val) => match val.parse::<u32>() {
-            Ok(num) => num,
-            Err(_) => {
-                eprintln!("BLAZE_PAGE_RESOURCE_COUNT not set. Using default value.");
-                100 // default value if parsing failsf
-            }
+
+    info!("fhir2sql started"); //@todo: replace with proper banner
+    
+    let pg_url = format!("postgresql://{}:{}@{}:{}/{}", 
+        config.pg_username, 
+        config.pg_password, 
+        config.pg_host, 
+        config.pg_port, 
+        config.pg_dbname);
+    let pg_con_pool = get_pg_connection_pool(&pg_url, 10).await?;
+    
+    info!("Running initial sync");
+    match run_sync(&pg_con_pool, &config).await {
+        Ok(()) => {
+            info!("Sync run successfull");
         },
-        Err(_) => {
-            eprintln!("BLAZE_PAGE_RESOURCE_COUNT not set. Using default value.");
-            100 // default value if env var is not set
+        Err(err) => {
+            error!("Sync run unsuccessfull: {}", err);
         }
-    };
-
-    let con_str = &format!("host={} port={} user={} password={} dbname={}", host, port, username, password, dbname);
-    let (mut client, connection) =
-    tokio_postgres::connect(&con_str, NoTls).await?;
+    }
 
-    tokio::spawn(async move {
-        if let Err(e) = connection.await {
-            //eprintln!("connection error: {}", e);
-            error!("Could not connect to PostgreSQL: {e}");
-        }
-    });
+    // main loop
+    info!("Entering regular sync schedule");
+    let mut interval = interval(Duration::from_secs(60)); // execute every 1 minute
     
-    sync(&blaze_base_url, &mut client, type_arg, batch_size, page_resource_count).await?;
+    loop {
+        select! {
+            _ = interval.tick() => {                  
+                let now = Utc::now().naive_local().time();                                
+                if now.hour() == config.target_time.hour() && now.minute() == config.target_time.minute() {
```

Comment:
There is also `tokio::time::sleep_until` which seems more like what you want although it may be a bit annoying to create the initial `Instant`

Reply:
I fiddled around with tokio::time::sleep_until a bit... don't remember the exact reason i did not end up with it.